topic-controller/src/main/java/io/strimzi/controller/topic/Config.java

```diff
@@ -34,6 +34,14 @@ public Long parse(String s) {
         }
     };
 
+    /** A Java Integer */
+    private static final Type<? extends Integer> INTEGER = new Type<Integer>() {
```

Comment:
What we actually need here is a (strictly?) positive integer. So maybe the `parse()` method should check for negatives (and zero?) and this `Type` be renamed `POSITIVE_INTEGER`?

Reply:
This makes be notice that in BackOff we're using `assert` for these bounds checks. We need to either ensure the TC (and CC) run with `-ea` or we should change them into regular `if/throw` checks.

Reply:
Maybe it's simpler for us having the logic encapsulated into the source code other than the `-ea` flag on the command line for launching the applications ? Wdyt ?

Reply:
Yeah, I agree it's better to use a regular `if`. Let's open an issue for this, and amend the checkstyle rules at the same time.

Reply:
I have created the #312 for that.